x-pack/plugin/relevance-search/src/main/java/org/elasticsearch/xpack/relevancesearch/xsearch/action/XSearchTransportAction.java

```diff
@@ -59,6 +61,8 @@ protected void doExecute(Task task, XSearchAction.Request request, ActionListene
                 l.onResponse(searchResponse);
             } catch (Exception t) {
                 l.onFailure(t);
+            } finally {
+                xSearchAnalyticsService.recordEvent(request, client);
```

Comment:
Nit: Maybe we could add the response as well to allow for storing response related analytics (like number of hits, time spent, etc).

Reply:
I like that suggestion, but I'd like to do it in a subsequent PR if you're good with that!